var/spack/repos/builtin/packages/vtk-m/package.py

```diff
@@ -187,3 +188,175 @@ def cmake_args(self):
                 options.append("-DVTKm_ENABLE_TBB:BOOL=OFF")
 
             return options
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    def check_install(self):
+        # default post installation check
+        # do nothing for now
+        pass
+
+    @run_after('install')
+    @on_package_attributes(run_tests=True)
+    @when('@master')
```

Comment:
I'm honestly not sure if stringing together 3 decorators works correctly. @alalazo?

Reply:
I need to read the code more in details if this doesn't work, so in case ping me again on that, but looking briefly I would expect that ordering matters and that:
```python
@run_after('install') # Register the object to be run after install
@when('@master') # Wraps the function in a new object with a local cache of methods
@on_package_attributes(run_tests=True) # Adds a conditional to a method
```
is a better ordering.

Reply:
After a few tests I fount that both ways of ordering decorators: 
```
@run_after('install')
@when('@master')
@on_package_attributes(run_tests=True)
```
and 
```
@run_after('install')
@on_package_attributes(run_tests=True)
@when('@master')
```
needs me to have a dummy test function to make the installation without errors.

I figured out that this order can make it right without needing to add a dummy function.
```
@when('@master')
@run_after('install')
@on_package_attributes(run_tests=True)
```
Feel free to let me know if you have any suggestions.

Reply:
I would keep `run_after` at the top. As I said I need to carefully check the current implementation, but I think the correctness of this ordering may rely on having called `when` before these three lines on a function named the same. 

Said otherwise: `when` potentially returns a different object than the one passed in, `run_after` registers the object passed in for execution at a specific time. It's probably more robust to register objects after they have been fully modified by other decorators.

Reply:
Ok. That makes sense. I have reorganized the smoke test function to keep `run_after` at the top and avoid using the dummy function. Please let me know if I need to modify any part.